buildSrc/subprojects/integration-testing/src/main/kotlin/org/gradle/gradlebuild/test/integrationtests/DistributionTest.kt

```diff
@@ -63,6 +66,30 @@ open class DistributionTest : Test() {
         jvmArgumentProviders.add(libsRepository)
         systemProperty("java9Home", project.findProperty("java9Home") ?: System.getProperty("java9Home"))
     }
+
+    override fun executeTests() {
+        setTimeoutMonitor()
+        super.executeTests()
+    }
+
+    private
+    fun setTimeoutMonitor() {
+        Timer(true).schedule(timerTask {
+            project.javaexec {
```

Comment:
❌ Should we make sure that the build is still running before trying to run `javaexec`? I guess this process runs on the daemon, so it will survive between builds, right?

Maybe we should even stop the timer(s) at the end of the build. WDYT? Should we only start the timers on `CI`?

Reply:
Good catch! Now after the `Test` finishes, the timer will be cancelled.